### DISCUSSION TITLE
add front facing camera usability

### DISCUSSION THREAD

**Discussion by gyroballmaster** [selfcustody/krux#216] [Ideas]
in case the rear camera has issues or doesn't work completely it would be nice to have a backup option. My unit came with a defective rear camera and has trouble scanning qr codes clearly now

**Comment by odudex** [comment]
It's a good idea. Frontal camera is good to read QR codes too(it's the default camera in Maix Dock), I don't know if the cable length is the same, but maybe they could be physically switched.
What kind of defect does it have? If it is focus, it can be adjusted by rotating the lenses carefully with the help of pliers.
You can share photos of it on [Telegram group](https://t.me/SC_Krux), we can try to help.

**Comment by odudex** [comment]
> in case the rear camera has issues or doesn't work completely it would be nice to have a backup option. My unit came with a defective rear camera and has trouble scanning qr codes clearly now

It seems some recently shipped Amigos had issues with main camera, we made a change in firmware(b10178e9be0a534508fef9792ea42aabcab825a9) that solved the issue for a similar case reported in Telegram, please try it out with [beta firmware](https://github.com/odudex/krux_binaries):
